src/main/java/io/vertx/ext/hawkular/impl/InventoryReporter.java

```diff
@@ -0,0 +1,241 @@
+ *  and Apache License v2.0 which accompanies this distribution.
+ *
+ *  The Eclipse Public License is available at
+ *  http://www.eclipse.org/legal/epl-v10.html
+ *
+ *  The Apache License v2.0 is available at
+ *  http://www.opensource.org/licenses/apache2.0.php
+ *
+ *  You may elect to redistribute this code under either of these licenses.
+ */
+package io.vertx.ext.hawkular.impl;
+
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.http.HttpClientOptions;
+import io.vertx.core.http.HttpClientResponse;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import io.vertx.ext.hawkular.AuthenticationOptions;
+import io.vertx.ext.hawkular.VertxHawkularOptions;
+
+import java.io.UnsupportedEncodingException;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.stream.Collectors.*;
+
+/**
+ * Report inventory to the Hawkular server.
+ *
+ * @author Austin Kuo
+ */
+public class InventoryReporter {
+  private static final Logger LOG = LoggerFactory.getLogger(InventoryReporter.class);
+
+  private static final CharSequence MEDIA_TYPE_APPLICATION_JSON = HttpHeaders.createOptimized("application/json");
+  private static final CharSequence HTTP_HEADER_HAWKULAR_TENANT = HttpHeaders.createOptimized("Hawkular-Tenant");
+
+  private final Vertx vertx;
+  private final String inventoryURI = "/inventory/deprecated";
+
+  private final CharSequence tenant;
+  private final CharSequence auth;
+
+  private final Map<CharSequence, Iterable<CharSequence>> httpHeaders;
+
+  private HttpClient httpClient;
+
+  private final String feedId;
+  private final String vertxRootResourceTypeId = "rt.vertx-root";
+  private final String eventbusResourceTypeId = "rt.eventbus";
+  private final String vertxRootResourceId;
+  private final String eventbusResourceId;
+
+  private final String counterMetricTypeId = "mt.counter";
+  private final String eventbusMetricId = "m.handlers";
+
+  private final int collectionInterval;
+  private final VertxHawkularOptions options;
+
+  /**
+   * @param vertx   the {@link Vertx} managed instance
+   * @param options Vertx Hawkular options
+   * @param context the metric collection and sending execution context
+   */
+  public InventoryReporter(Vertx vertx, VertxHawkularOptions options, Context context) {
+    this.vertx = vertx;
+    this.options = options;
+    feedId = options.getFeedId();
+    vertxRootResourceId = options.getVertxRootResourceId();
+    eventbusResourceId = vertxRootResourceId + ".eventbus";
+    collectionInterval = options.getSchedule();
+
+    tenant = options.isSendTenantHeader() ? HttpHeaders.createOptimized(options.getTenant()) : null;
```

Comment:
All the code below is taken from the Sender, correct? It would be better to factor it out. Maybe even share the HttpClient instance (since it is used by the same Context).